pkg/metric/prometheus/registry.go

```diff
@@ -0,0 +1,27 @@
+package prometheus
+
+import (
+	"context"
+	"sync"
+
+	"github.com/justtrackio/gosoline/pkg/appctx"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/collectors"
+)
+
+var registryLock sync.Mutex
```

Comment:
I can see why one would need a lock here but ideally this would be handled by the appctx.Provide function, right? I was a little bit surprised that that one has no locking yet..  
If you feel like it, you could try to add a locking mechanism there which locks based on the key? :thinking:

Reply:
makes sense, could solve potential future issues, in my case because I want to access the registry almost at the same time I had issues. Moved the lock to appctx. 

I had one issue though, I wanted to create a KeyLock in `conc` pkg, but because that pkg has dependency on `ddb` which uses the `metric` pkg I had an import cycle, I would like probably to move all the ddb related locks inside a sub pkg, but that would cause breaking changes, what's your opinion? currently this lock implementation is in appctx

Reply:
We can keep the lock implementation here for now..  
Maybe just move it into a dedicated file to make it more prominent?  
I'm not a big fan of mixing to many "complex" types in a single file..

Reply:
I actually want it to be in `conc` pkg but to do so I'll have to move the locks that have the `ddb` dependency somewhere else, what do you think?

Reply:
Where do you would put it?